### PR TITLE
Canonicalizations for dispatch.tensor.load

### PR DESCRIPTION
**Pull request by asaadaldien**
- Adds a builder to `dispatch.tensor.load` to accept mixedOperands.
- Canonicalizes `dispatch.tensor.load` operands with upstream pattern.

### CODE REVIEW

File: iree/compiler/Dialect/Flow/IR/FlowOpFolders.cpp [hunk 1]
```diff
@@ -429,11 +429,74 @@ struct ConvertDispatchInputLoadOfTensorToSubTensor
...
+
+// Pulls in constant index operands as static arguments and return type with the
+// new static sizes.
+struct CanonicalizeStaticOperands
+    : public OpRewritePattern<DispatchTensorLoadOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(DispatchTensorLoadOp loadOp,
+                                PatternRewriter &rewriter) const override {
+    if (loadOp.strides().empty() || loadOp.sizes().empty() ||
```
Comment (MaheshRavishankar) [inline iree/compiler/Dialect/Flow/IR/FlowOpFolders.cpp:441]: This could be cleaned up a bit if you could have a builder method for `dispatch.tensor.load` that takes `ArrayRef<OpFoldResult>` for offsets, sizes and strides.

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Relatively minor recommendation. THanks!

File: iree/compiler/Conversion/LinalgToLLVM/LLVMCodeGenOptions.cpp [hunk 2]
```diff
@@ -24,23 +24,29 @@ static llvm::cl::opt<bool> clEnableLLVMLinalgOnTensors(
...
                    "linag.matmul"),
     llvm::cl::init(false));
 
-static llvm::cl::opt<bool> unfusedFMA(
+static llvm::cl::opt<bool> clUnfusedFMA(
     "iree-codegen-linalg-to-llvm-use-unfused-fma",
     llvm::cl::desc("Enable rewriting llvm.fma to its unfused version."),
     llvm::cl::init(false));
 
+static llvm::cl::opt<bool> clEnableLinalgOnTensorsToVectors(
```
Comment (MaheshRavishankar) [inline iree/compiler/Conversion/LinalgToLLVM/LLVMCodeGenOptions.cpp:39]: These seem unrelated. Can you split them into a different PR?

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
One more minor nit: Some changes are unrelated to this. Better to have that in a separate PR if possible.

File: iree/compiler/Conversion/LinalgToLLVM/LLVMCodeGenOptions.cpp (hunk 2 above)
Comment (asaadaldien) [reply iree/compiler/Conversion/LinalgToLLVM/LLVMCodeGenOptions.cpp:39]: This PR will fail because now vectorization on tensors pass will kick in for statistically shaped tensors and blow up for big tiles.

Comment (MaheshRavishankar) [reply iree/compiler/Conversion/LinalgToLLVM/LLVMCodeGenOptions.cpp:39]: Ah, now I understand. Ok thanks! In reality, should just remove that pass. It is actually not used right now.

**Review by MaheshRavishankar — APPROVED** [review]

File: iree/compiler/Conversion/LinalgToLLVM/LLVMCodeGenOptions.cpp (hunk 2 above)
Comment (asaadaldien) [reply iree/compiler/Conversion/LinalgToLLVM/LLVMCodeGenOptions.cpp:39]: Correct! this pass with no-op before this PR. We can keep it for now and revisit once explored tensor -> vector pass. I think for matmul if we did the tiling in a separate pass before this pass this one will stay as is.

Comment (MaheshRavishankar) [reply iree/compiler/Conversion/LinalgToLLVM/LLVMCodeGenOptions.cpp:39]: Sure. Approved this PR already.
